core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -162,45 +156,40 @@ public void doWork() {
             this.storeUpdateHandler.handleUpdate(messageKey, message);
             try {
               offsetUpdateLock.lock();
-              offsetInSchemasTopic = messageAndMetadata.offset();
+              offsetInSchemasTopic = record.offset();
               offsetReachedThreshold.signalAll();
             } finally {
               offsetUpdateLock.unlock();
             }
           } catch (StoreException se) {
-            log.error("Failed to add record from the Kafka topic" + topic + " the local store");
+            throw new RuntimeException("Failed to add record from the Kafka topic" + topic + " the local store");
           }
         }
       }
-    } catch (ConsumerTimeoutException cte) {
-      // Expect ConsumerTimeout == -1, so this *should* never happen
-      throw new IllegalStateException(
-          "KafkaStoreReaderThread's ConsumerIterator timed out despite expected infinite timeout.");
-    } catch (MessageSizeTooLargeException mstle) {
+    } catch (WakeupException we) {
+      consumer.close();
```

Comment:
Ah, that makes sense. I forget we were using ShutdownableThread. In that case, I wonder if it makes sense to catch and ignore the WakeupException and then call close() in the shutdown() hook after the call to `super.shutdown()`. That ensures that the loop has actually exited before close() is called. Otherwise it seems we have a couple race conditions where it's possible that we either 1) continue using the consumer after it has been closed, or 2) never call close() at all.